ci/lint/format.sh

```diff
@@ -266,26 +238,11 @@ format_files() {
 }
 
 format_all_scripts() {
-    command -v flake8 &> /dev/null;
-    HAS_FLAKE8=$?
-
-    # Run isort before black to fix imports and let black deal with file format.
-    echo "$(date)" "isort...."
-    git ls-files -- '*.py' "${GIT_LS_EXCLUDES[@]}" | xargs -P 10 \
-      isort
     echo "$(date)" "Black...."
     git ls-files -- '*.py' "${GIT_LS_EXCLUDES[@]}" | xargs -P 10 \
       black "${BLACK_EXCLUDES[@]}"
     echo "$(date)" "MYPY...."
     mypy_on_each "${MYPY_FILES[@]}"
-    if [ $HAS_FLAKE8 ]; then
-      echo "$(date)" "Flake8...."
-      git ls-files -- '*.py' "${GIT_LS_EXCLUDES[@]}" | xargs -P 5 \
-        flake8 --config=.flake8
-
-      git ls-files -- '*.pyx' '*.pxd' '*.pxi' "${GIT_LS_EXCLUDES[@]}" | xargs -P 5 \
-        flake8 --config=.flake8 "$FLAKE8_PYX_IGNORES"
-    fi
```

Comment:
Don't we need to replace this with `ruff`?

Reply:
See my comment here https://github.com/ray-project/ray/pull/49435#discussion_r1897546678. These comments were folded.

Reply:
What about `*.py`?

Reply:
I modified the Ruff pre-commit hook to automatically locate the corresponding Python files. See https://github.com/ray-project/ray/pull/49435/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R34-R36 and https://github.com/astral-sh/ruff-pre-commit/blob/6299d2ca232741e7e33e7fd8a5a10059bd743da7/.pre-commit-hooks.yaml#L6 for details.

Reply:
Does CI run pre-commit hook or it only runs when people enabled pre-commit hook locally?

Reply:
CI runs pre-commit hook here:

- https://buildkite.com/ray-project/premerge/builds/32538#01940ab3-1b33-4d80-ba3c-a62c82ed7f88
- https://github.com/ray-project/ray/pull/49435/files#diff-ad6658a2c0375932b4a9f18312962089115b0d5a0c817e84e0213cc78cbfe927R17-R18